exercises/spectre/descr.md

```diff
@@ -0,0 +1,299 @@
+# From a Spectre to a Tree
+
+Suppose we have an algebraic data type `'a tree` of binary trees where each
+leaf carries a piece of data of type `'a`.
+
+```
+  type 'a tree =
+    | Leaf of 'a
+    | Fork of 'a tree * 'a tree
+```
+
+For instance, the tree `Fork (Leaf 1, Fork (Leaf 2, Leaf 3))` has type `int
+tree`, as its leaves carry integers.
+The list of the data values carried by its leaves is `[1; 2; 3]` and the list
```

Comment:
It could have been clearer with a `char tree` instead of an `int tree`.

Reply:
I am changing the initial example to use characters instead of integers.